packages/mds-cache/redis/index.ts

```diff
@@ -0,0 +1,168 @@
+import Redis, { KeyType, ValueType } from 'ioredis'
+import { Nullable, Timestamp } from '@mds-core/mds-types'
+import { isDefined, ClientDisconnectedError, ExceptionMessages } from '@mds-core/mds-utils'
+import { initClient } from './helpers/client'
+import { OrderedFields } from '../@types'
+
+// /////////////////// start back-ported junk
+
+export const RedisCache = () => {
+  let client: Nullable<Redis.Redis> = null
+
+  /**
+   * If the client is defined, the closure is called, otherwise throws an error
+   * @param closure called with a Redis client, returns the result
+   * @returns same as what the closure returns
+   * @throws ClientDisconnectedError
+   */
+
+  const safelyExec = async <T>(closure: (theClient: Redis.Redis) => T) => {
+    if (isDefined(client)) {
+      return closure(client)
+    }
+    throw new ClientDisconnectedError(ExceptionMessages.INITIALIZE_CLIENT_MESSAGE)
+  }
+
+  return {
+    initialize: async () => {
+      client = await initClient()
+    },
+    shutdown: async () => {
+      if (isDefined(client)) {
+        client.disconnect()
+      }
+      client = null
+    },
+    multi: async () => {
+      return safelyExec(theClient => {
+        return theClient.multi()
+      })
+    },
+    get: async (key: KeyType) => {
```

Comment:
What was the motivation for removing the _Async_ suffix convention? `get` vs `getAsync`, etc.?

Reply:
It was entirely an artifact of Bluebird.